### PR TITLE
Synthesis stage: support PWG neural vocoder & GL

### PR DESCRIPTION
**Pull request by unilight**
This is PR for synthesis (w/ neural vocoder) support. 

- [x] implement GL

1. I put the pretrained model in google drive, and I used the download script from ESPnet. Is it okay...?

2. In the current implementation, each conversion pair is generated in stage 5, so in the case of VCC2018, there are `(# spks) x (# spks) x (# eval utts)=12*12*35=5040` utterances to normalize and decode, which takes about 15 mins on a single GeForce GTX 1080 Ti. I think it is a bit too long. Should we add `srcspk` and `trgspk` and decode/synthesize one pair at a time?

3. Shall we also decode/synthesize the dev set?

### CODE REVIEW

**Comment by k2kobayashi** [comment]
Good work!!
1. The license of ESPNet code is Apache 2.0. It's fine.
2. We can control it using ```eval/wav.scp```. And, I think 15 minutes are not so long. In terms of challenge dataset it has been defined its evaluation set. So, it should retain its structure.  
3. For the stage 6, I think it is not necessary. But I'm happy to decode while training (i.e., on stage 3) if possible. 

By the way, could you share resulting sounds using PWG?

File: egs/vaevc/vcc2018v1/local/pretrained_model_download.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+case "${pretrained_model}" in
+    "PWG")             share_url="https://drive.google.com/open?id=1N9xqzRte6SGP6ZpPNS7uug8uCkM-sCeB" ;;
+    *) echo "No such pretrained model: ${pretrained_model}"; exit 1 ;;
+esac
+
+dir=${download_dir}
+mkdir -p ${dir}
+if [ ! -e ${dir}/.complete ]; then
+    utils/download_from_google_drive.sh ${share_url} ${dir} ".tar.gz"
+    touch ${dir}/.complete
```
Comment (k2kobayashi) [inline egs/vaevc/vcc2018v1/local/pretrained_model_download.sh]: ```.done``` it better.

File: egs/vaevc/template/run.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@
 #  4: reconstruction
 #  5: evaluation
```
Comment (k2kobayashi) [inline egs/vaevc/template/run.sh]: Please write comment here.

**Review by k2kobayashi — APPROVED** [review]
After we implement GL and world vocoder decoding and test them for vcc2018 and vcc2020, we will create new release!

**Comment by unilight** [comment]
> 3. For the stage 6, I think it is not necessary. But I'm happy to decode while training (i.e., on stage 3) if possible.

Currently, I am using the normalization/decoding scripts of `parallel_wavegan`, so it might not be so straightforward to combine into the training procedure. Maybe I will try in the future.

> By the way, could you share resulting sounds using PWG?

Sure. https://drive.google.com/file/d/1L7TYFcwOwV_e-8UJUOkw-dEUMARlkdUz/view?usp=sharing
In terms of quality, I think the performance of PWG is not so good. I am tuning the model.
On the other hand, the similarity is also not so good (comparing to my experience of applying normal VAE-VC to VCC2018).

File: egs/vaevc/template/run.sh [hunk 3]
```diff
@@ -33,6 +34,12 @@ featsscp="None"
 conf=conf/mlfb_vqvae.yml  # newtork config
 spkr_yml=conf/spkr.yml # speaker config
 
+# synthesis related
+model_step=                     # If not specified, use the latest.
```
Comment (k2kobayashi) [inline egs/vaevc/template/run.sh:38]: voc_model_step would be better.

File: egs/vaevc/template/run.sh [hunk 4]
```diff
@@ -152,5 +159,69 @@ if [ "${stage}" -le 5 ] && [ "${stop_stage}" -ge 5 ]; then
...
+    [ -z "${model_step}" ] && model_step="$(find "${expdir}/${confname}" -name "*.pkl" -print0 \
+        | xargs -0 ls -t | head -n 1 | cut -d"_" -f 2 | cut -d"s" -f 1)"
+    outdir=${expdir}/${confname}/eval_${voc}_wav/${model_step}
+    outwavdir=${outdir}/wav; mkdir -p ${outwavdir}
+
+    # GL
+    if [ ${voc} = "GL" ]; then
+        echo "Using Griffin-Lim phase recovery."
+        ${train_cmd} "${outwavdir}/decode.log" \
+            utils/griffin-lim.py \
```
Comment (k2kobayashi) [inline egs/vaevc/template/run.sh]: How about placing to ```crank/bin```?

**Review by k2kobayashi — COMMENTED** [review]
Good work. I have a few comments only.

**Comment by k2kobayashi** [comment]
Thanks for the sounds. I think the quality is reasonable but I hope you'll make some improvements.

File: egs/vaevc/template/run.sh (hunk 3 above)
Comment (unilight) [reply egs/vaevc/template/run.sh:38]: The `model_step` here is to specify which conversion model checkpoint to use, not the vocoder.

**Comment by unilight** [comment]
> Thanks for the sounds. I think the quality is reasonable but I hope you'll make some improvements.

Do you mean improving the PWG or the conversion model?

**Comment by k2kobayashi** [comment]
Hopefully both of them. Anyway, it's better to add objective evaluation scripts to optimize hyper-parameter.
